### PR TITLE
Fix trivially fixable warnings

### PR DESCRIPTION
**Pull request by Ozaq**
When building on gcc9 and clang 8 there where a ton of warnings. I have fix all trivially fixable ones on this PR. (Note this does not build on clang8 w.o. my other PR)
---
Edit: Also I did not yet clear tests from warnings!
---

There are 3 warnings left that cannot be addressed w.o. having a deeper look into it:
1. WaitingArea.cpp:
```
[19/97] Building CXX object CMakeFiles/core.dir/geometry/WaitingArea.cpp.o
/mnt/fast/projects/jpscore/geometry/WaitingArea.cpp:147:27: warning: comparison of integers of different signs: 'std::set<int, std::less<int>, std::allocator<int> >::size_type' (aka 'unsigned long') and 'int' [-Wsign-compare]
     if (pedInside.size() >= maxNumPed){
         ~~~~~~~~~~~~~~~~ ^  ~~~~~~~~~
/mnt/fast/projects/jpscore/geometry/WaitingArea.cpp:155:27: warning: comparison of integers of different signs: 'std::set<int, std::less<int>, std::allocator<int> >::size_type' (aka 'unsigned long') and 'int' [-Wsign-compare]
     if (pedInside.size() < maxNumPed){
         ~~~~~~~~~~~~~~~~ ^ ~~~~~~~~~
/mnt/fast/projects/jpscore/geometry/WaitingArea.cpp:158:27: warning: comparison of integers of different signs: 'std::set<int, std::less<int>, std::allocator<int> >::size_type' (aka 'unsigned long') and 'int' [-Wsign-compare]
     if (pedInside.size() < minNumPed){
         ~~~~~~~~~~~~~~~~ ^ ~~~~~~~~~
/mnt/fast/projects/jpscore/geometry/WaitingArea.cpp:176:28: warning: comparison of integers of different signs: 'std::set<int, std::less<int>, std::allocator<int> >::size_type' (aka 'unsigned long') and 'int' [-Wsign-compare]
     if ((pedInside.size() >= minNumPed) && (startTime < 0. )){
```
The problem here is that the initial state of `minNumPed` and `maxNumPed` is -1 and during a signed unsigned comp. this will be cast into a signed int i.e. on a freshly constructed object of WaitingArea all those checks will behave unexpected!

2. EventManager.cpp 
```
/mnt/fast/projects/jpscore/events/EventManager.cpp:528:24: warning: enumeration value 'Error' not handled in switch [-Wswitch]
               switch (event.GetState()){
                       ^
1 warning generated.
```
Here the `Error` state is no handled, I did not try to fix this because I might unknowingly change the logic. this is better left to someone who knows the code.

3. tinyxmlparser.cpp
```
/mnt/fast/projects/jpscore/tinyxml/tinyxmlparser.cpp:113:10: warning: this statement may fall through [-Wimplicit-fallthrough=]
  113 |    input >>= 6;
      |    ~~~~~~^~~~~
/mnt/fast/projects/jpscore/tinyxml/tinyxmlparser.cpp:114:3: note: here
  114 |   case 3:
      |   ^~~~
/mnt/fast/projects/jpscore/tinyxml/tinyxmlparser.cpp:117:10: warning: this statement may fall through [-Wimplicit-fallthrough=]
  117 |    input >>= 6;
      |    ~~~~~~^~~~~
/mnt/fast/projects/jpscore/tinyxml/tinyxmlparser.cpp:118:3: note: here
  118 |   case 2:
      |   ^~~~
/mnt/fast/projects/jpscore/tinyxml/tinyxmlparser.cpp:121:10: warning: this statement may fall through [-Wimplicit-fallthrough=]
  121 |    input >>= 6;
      |    ~~~~~~^~~~~
/mnt/fast/projects/jpscore/tinyxml/tinyxmlparser.cpp:122:3: note: here
  122 |   case 1:
      |   ^~~~
```
yeah the comment above the code says it all :) `scary scary fall trough`, I probably would try to replace this with libicu to convert between UTF8 and UTF32 instead of doing that myself.

Let me know if this is helpful.

Cheers

### CODE REVIEW

File: geometry/Point.cpp [hunk 1]
```diff
@@ -32,16 +32,6 @@
 /************************************************************
   Konstruktoren
  ************************************************************/
-
-Point::Point(const Point& orig)
```
Comment (chraibi) [inline geometry/Point.cpp]: Why is this constructor removed?

File: geometry/Line.h [hunk 2]
```diff
@@ -60,9 +60,7 @@ class Line {
 
      Line(const Point& p1, const Point& p2);
 
-     Line(const Line& orig);
```
Comment (chraibi) [inline geometry/Line.h]: why is this constructor removed?

File: pedestrian/Ellipse.cpp [hunk 3]
```diff
@@ -55,21 +55,6 @@ JEllipse::JEllipse()
      _vel0 = 0; // desired speed
 }
 
-JEllipse::JEllipse(const JEllipse& orig)
```
Comment (chraibi) [inline pedestrian/Ellipse.cpp]: maybe the same logic as before, but again.. this constructor is just removed?

**Comment by Ozaq** [comment]
@chraibi The constructors above have been removed because the classes they are part of are trivially copyable, and if you define a custom Copy Ctor you also need to define a copy assignment operator. So instead of also implementing the copy assignment you can simply remove the copy ctor.

**Pull request merged**